meldingen_core/actions/melding.py

```diff
@@ -244,7 +244,36 @@ async def __call__(self, melding_id: int, token: str) -> Sequence[A]:
         return await self._answer_repository.find_by_melding(melding_id)
 
 
-class MeldingSubmitAction(BaseMeldingFormStateTransitionAction[T]):
+class MeldingSubmitAction(BaseCRUDAction[T]):
+    _repository: BaseMeldingRepository[T]
+    _state_machine: BaseMeldingStateMachine[T]
+    _verify_token: TokenVerifier[T]
+    _invalidate_token: BaseTokenInvalidator[T]
+
+    def __init__(
+        self,
+        repository: BaseMeldingRepository[T],
+        state_machine: BaseMeldingStateMachine[T],
+        token_verifier: TokenVerifier[T],
+        token_invalidator: BaseTokenInvalidator[T],
+    ) -> None:
+        self._repository = repository
+        self._state_machine = state_machine
+        self._verify_token = token_verifier
+        self._invalidate_token = token_invalidator
+
+    async def __call__(
+        self,
+        melding_id: int,
+        token: str,
+    ) -> T:
+        melding = await self._verify_token(melding_id, token)
+        await self._state_machine.transition(melding, self.transition_name)
+        # await self._repository.save(melding)
```

Comment:
MIsschien beter om deze line als laatst uit te voeren en niet meer vanuit de invalidator.

Reply:
Yes, dit is misschien toch het beste. Zo werkt het nu ook op de meeste plekken. 

Ergens voelt het gek om je wijzigingen niet in je class op te slaan, maar het voelt ook onnodig om dat dan weer bij elke aparte class te doen.